src/target/riscv/riscv-013.c

```diff
@@ -145,6 +145,60 @@ typedef struct {
 	struct target *target;
 } target_list_t;
 
+struct ac_cache {
+	uint32_t *commands;
+	size_t size;
+};
+
+static int ac_cache_elem_comparator(const void *lhs, const void *rhs)
+{
+	return *(const uint32_t *)rhs - *(const uint32_t *)lhs;
+}
```

Comment:
There seems to be a hidden bug in this comparator:

- If `int` is 32 bits, this function will work as expected (but there is a hidden implicit conversion from uint32 to int that can change the sign of the result, and gcc would complain if `-Wconversion` is enabled).
- If `int` type is 64 bits wide, then this function would never return a negative value. (Also we would probably not get any `-Wconversion` notification.)

For that reason, please consider re-writing using if + comparison instead of subtraction:

```suggestion
static int ac_cache_elem_comparator(const void *p_lhs, const void *p_rhs)
{
	unint32_t lhs = *(const uint32_t *)p_lhs;
	unint32_t rhs = *(const uint32_t *)p_rhs;
	
	if (lhs < rhs)
		return -1;
	if (rhs > lhs)
		return 1;	
	return 0;
}
```

Reply:
Thanks for the catch! Addressed